Added new route for profile plan type

jayeshmangwani opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Explanation of Change
<!-- Explain what your change does and how it addresses the linked issue -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/51768
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Go to Workspaces -> Workspace -> Profile.
2. If the user has admin access, they will see the Plan Type tab under the Company address menu item.
3. Click on Plan Type to open a new RHP screen for Plan Type, where the current workspace plan will be preselected.

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image

It's acceptable to write "Same as tests" if the QA team is able to run the tests in the above "Tests" section.
--->
// TODO: These must be filled out, or the issue title must include "[No QA]."

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/670f30b6-10ec-4aa8-9073-703b02665008


<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b474a0a7-e58f-4ff2-94b1-7d2b12d8b81d


<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/0a093048-18c7-458a-9590-5cabf5192b0b


<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/a1e67340-cd74-4d85-a307-f52c0b984a46


<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/cafeda4a-427d-47ed-8a16-977254be125a


<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/aa13bbc3-c7f9-4d37-82d3-e17e9f797bfc


<!-- add screenshots or videos here -->

</details>

jayeshmangwani committed:
added new route for profile plan type

jayeshmangwani committed:
merge main and resolved conflicts

jayeshmangwani committed:
added plan type and pricing help for URL

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
added selection list for ws plan type selection

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani commented:
In this PR, @DylanDylann , we’ve added the Plan Type row and a new RHP page specifically for Plan Type. Currently, we're only implementing the pre-selection of the Plan Type on this page.

jayeshmangwani committed:
removed unused en.ts key

jayeshmangwani committed:
removed unused es.ts key

DylanDylann commented:
BUG: Offline indicator displays above the button

<img width="324" alt="Screenshot 2024-11-12 at 11 30 52" src="https://github.com/user-attachments/assets/8afe584e-e19f-45db-b8af-3b766c5fcea5">

DylanDylann commented:
BUG: 
- The button should display above the device border
- The subtitle should be in the scroll view

https://github.com/user-attachments/assets/8e3a4139-1c18-4243-ae92-9d88816b27bb

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
fixed footer button position when offline

jayeshmangwani commented:
Thanks for the review, @DylanDylann! I’ve fixed the button position issue when offline. I’ll be adding the upgrade component to this PR next.
![Simulator Screenshot - iPhone 15 Pro - 2024-11-12 at 18 46 22](https://github.com/user-attachments/assets/d55e161e-f3ed-46b1-9375-5c38022512c4)

jayeshmangwani commented:
> The subtitle should be in the scroll view

We’re currently keeping the subtitle text outside the ScrollView for the SelectionList on some pages, like the Priority and Theme pages. I don’t have a strong opinion on this; I’ve kept it outside mainly for consistency.

https://github.com/user-attachments/assets/f09d70d7-2a30-41cc-881b-8e3465f4b28a

DylanDylann commented:
@jayeshmangwani Thanks for your update. Please ping me when you push the new code for upgrade component

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
added locked plan type UI to plan type page

jayeshmangwani committed:
made featureName optional and added a common features UI for the upgrade.

jayeshmangwani committed:
merge main , resolved conflicts

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
updated common upgrade intro UI styling same as other intro components

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
fixed lint

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

DylanDylann commented on src/CONST.ts:
```diff
@@ -931,6 +931,7 @@ const CONST = {
     CONFIGURE_REIMBURSEMENT_SETTINGS_HELP_URL: 'https://help.expensify.com/articles/expensify-classic/workspaces/Configure-Reimbursement-Settings',
     COPILOT_HELP_URL: 'https://help.expensify.com/articles/expensify-classic/copilots-and-delegates/Assign-or-remove-a-Copilot',
     DELAYED_SUBMISSION_HELP_URL: 'https://help.expensify.com/articles/expensify-classic/reports/Automatically-submit-employee-reports',
+    PLAN_TYPES_AND_PRICING_HELP_URL: 'https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/Plan-types-and-pricing',
```
Add my review

jayeshmangwani replied:
Used URL as per this comment https://github.com/Expensify/App/issues/51494#issuecomment-2462110622

DylanDylann replied:
No, my comment is to add this PR to my K2s List. No problem here

jayeshmangwani committed:
updated spanish copy for locked type subscriptions

jayeshmangwani commented on src/ROUTES.ts:
```diff
@@ -965,9 +969,9 @@ const ROUTES = {
-            getUrlWithBackToParam(`settings/workspaces/${policyID}/upgrade/${encodeURIComponent(featureName)}` as const, backTo),
+        route: 'settings/workspaces/:policyID/upgrade/:featureName?',
+        getRoute: (policyID: string, featureName?: string, backTo?: string) =>
+            getUrlWithBackToParam(`settings/workspaces/${policyID}/upgrade/${encodeURIComponent(featureName ?? '')}` as const, backTo),
```
We're using the existing Upgrade page to display the upgrade workspace page when coming from the workspace plan page. Now, we can also navigate from pages that don't have a specific feature.

I've made featureName an optional parameter. If it's not provided, we'll show the common control plan features instead.

jayeshmangwani commented on src/languages/en.ts:
```diff
@@ -4343,6 +4358,23 @@ const translations = {
+            lockedPlanDescription: ({subscriptionUsersCount, annualSubscriptionEndDate}: WorkspaceLockedPlanTypeParams) =>
+                `You've committed to ${subscriptionUsersCount} active users on the Control plan until your annual subscription ends on ${annualSubscriptionEndDate}. You can switch to pay-per-use subscription and downgrade to the Collect plan starting ${annualSubscriptionEndDate} by disabling auto-renew in`,
+            subscriptions: 'Subscriptions',
+        },
```
these translations confirmed here https://expensify.slack.com/archives/C01GTK53T8Q/p1731048615013379?thread_ts=1730105986.836259&cid=C01GTK53T8Q

carlosmiceli replied:
Can we resolve this conversation?

jayeshmangwani commented on src/languages/en.ts:
```diff
@@ -4240,6 +4242,19 @@ const translations = {
+                    benefit3: 'Multiple approval workflows',
+                    benefit4: 'Enhanced security controls',
+                },
+            },
```
These translations are yet to be confirmed here: https://expensify.slack.com/archives/C01GTK53T8Q/p1731048615013379?thread_ts=1730105986.836259&cid=C01GTK53T8Q

jayeshmangwani replied:
@carlosmiceli Before merging, I wanted to let you know that these translations are not yet confirmed.

carlosmiceli replied:
Great, let's keep these comments open so we can go back to them to see what we have yet to change/merge.

jayeshmangwani commented on src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:
```diff
@@ -0,0 +1,128 @@
+    const isControl = policy?.type === CONST.POLICY.TYPE.CORPORATE;
+    const isAnnual = privateSubscription?.type === CONST.SUBSCRIPTION.TYPE.ANNUAL;
+
+    const isPlanTypeLocked = isControl && isAnnual;
```
`isPlanTypeLocked` is not the final logic here, as I'm currently unsure how to determine if a specific policy is on a yearly plan. I've asked for clarification https://expensify.slack.com/archives/C02NK2DQWUX/p1731682677445059, but for now, we can finalize the UI and adjust the show logic later.

jayeshmangwani replied:
Before merging, I also wanted to let you know that `isPlanTypeLocked` logic is also not yet confirmed

jayeshmangwani replied:
@carlosmiceli ⬆️

carlosmiceli replied:
Got it, if we can add it later then that's all good.

carlosmiceli replied:
I think there's a few relevant answers now in the slack thread :)

jayeshmangwani committed:
update en and es translation for benefits learn more

jayeshmangwani commented:
**Plan Types  and Workspace Upgrade Attachments**




<details>
  <summary>iOS</summary>

  https://github.com/user-attachments/assets/0a82dfb8-42c8-47f9-b4ef-b5b47d2f3115

 
</details>


<details>
  <summary>Desktop:</summary>
  
  https://github.com/user-attachments/assets/92cb7383-557f-4a10-80a7-d46a68fe3626

  
</details>

<details>
  <summary>web:</summary>
  
  https://github.com/user-attachments/assets/33337af7-0035-47bd-a093-99649c74e00d

  
</details>

jayeshmangwani commented:
@DylanDylann I've added the Plan Type page and handled the upgrade workspace case. Feel free to review! A few things are yet to be confirmed (I’ve added comments above for those), but they shouldn't block the initial review of these changes.

Thanks!

jayeshmangwani committed:
Merge branch 'main' into workspace_downgrade_plan_type_page

DylanDylann commented on src/pages/workspace/upgrade/UpgradeIntro.tsx:
```diff
@@ -88,4 +98,60 @@ function UpgradeIntro({feature, onUpgrade, buttonDisabled, loading}: Props) {
     );
 }
 
+function CommonCorporateFeatures({buttonDisabled, loading, onUpgrade}: {buttonDisabled?: boolean; loading?: boolean; onUpgrade: () => void}) {
```
@jayeshmangwani Let's move this component to a new file

DylanDylann commented:
@jayeshmangwani Also please help to resolve conflict

DylanDylann commented:
Please add pending action to the new field when creating a new workspace 

https://github.com/user-attachments/assets/317d69ca-04ab-4bda-85b8-2fa68a291a14

jayeshmangwani committed:
merged main and resolved conflicts

jayeshmangwani committed:
generic upgrade features moved to a new file and added pending action add for workspace type

jayeshmangwani commented:
Thanks for the review, @DylanDylann! I’ve resolved the merge conflicts and `CommonCorporateFeatures` migrated to a new component (GenericFeaturesView). However, I’m not sure if the new component name fully justifies its purpose 😄.

DylanDylann commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

DylanDylann commented:
@carlosmiceli Friendly bump

carlosmiceli commented:
Thanks for the bump, been waiting for a review on a BE PR to get merged. @DylanDylann correct me if I'm wrong, but wouldn't we risk showing a non-working plan type selector with this PR if the BE is not ready?

DylanDylann commented:
@carlosmiceli We can merge this PR first without an update from BE, this PR is about the upgrade flow and the API for this is done before

carlosmiceli commented:
Gotcha, let's solve conflicts and we can merge! 🙇 
cc @jayeshmangwani

jayeshmangwani committed:
merged main and resolved conflicts

jayeshmangwani commented:
Conflicts have been resolved @carlosmiceli. Before merging the PR, please check these comments: https://github.com/Expensify/App/pull/52117#discussion_r1860918653 and https://github.com/Expensify/App/pull/52117#discussion_r1860919403

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

carlosmiceli commented:
@jayeshmangwani how does it work with the changes from this PR that got reverted? Are you creating another PR with the same changes now that the BE is ready?

jayeshmangwani commented:
Yes, I'll be raising the PR in a few hours. I'm away from the keyboard right now.

carlosmiceli commented:
No rush at all, it's Saturday 😄

jayeshmangwani commented:
@carlosmiceli @DylanDylann New PR containing the same changes has been created here https://github.com/Expensify/App/pull/53343
